sdk/python/lib/pulumi/errors.py

```diff
@@ -12,9 +12,40 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from typing import List, Optional, TypedDict
+
 
 class RunError(Exception):
     """
     Can be used for terminating a program abruptly, but resulting in a clean exit rather than the usual
     verbose unhandled error logic which emits the source program text and complete stack trace.
     """
+
+
+class InputPropertyError(Exception):
+    def __init__(self, message: str, property_path: str, reason: str):
```

Comment:
A [question](https://github.com/pulumi/pulumi/pull/17325#discussion_r1781012031) for the Node.js API as well, but when would `message` be different than `reason`?

Should this take only a `property_path` and `reason` and create a message based on those?

Reply:
Done, same as in the Node.js API.